db.go

```diff
@@ -184,19 +197,24 @@ func (x *dummyUserStore) Authenticate(identity, password string) (er error) {
 		er = ErrInvalidPassword
 	}
 
-	return
+	return "", er
 }
 
 func (x *dummyUserStore) Close() {
 	//Set incrementing user id to 0, for unit test prediction
 	nextUserId = 0
 }
 
-func (x *dummyUserStore) SetPassword(userId UserId, password string, checkPasswordReuse bool) error {
+func (x *dummyUserStore) SetConfig(passwordExpiry time.Duration) error {
+	x.passswordExpiry = passwordExpiry
+	return nil
+}
+
+func (x *dummyUserStore) SetPassword(userId UserId, password string, authTypeCheck AuthCheck) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
 	if user, exists := x.users[userId]; exists && user.authUserType.CanSetPassword() {
-		if checkPasswordReuse && user.password == password {
+		if authTypeCheck&AuthCheckNonPasswordReuse == 1 && user.password == password {
```

Comment:
Firstly it does not look like this file is formatted.

And you went through the trouble of creating a new type but not using it like so:
`authTypeCheck == AuthCheckNonPasswordReuse`

Reply:
Right way to check for flags is: `!!(authTypeCheck&AuthCheckNonPasswordReuse)`. That works in C++. Not sure about Go. You might have to do `(authTypeCheck&AuthCheckNonPasswordReuse) != 0`